docs/csharp/tutorials/pattern-matching.md

```diff
@@ -0,0 +1,325 @@
+> * How to recognize situations where pattern matching should be used.
+> * How to use pattern matching expressions to implement behavior based on types and property values.
+> * How to combine pattern matching with other techniques to create complete algorithms.
+
+## Prerequisites
+
+You'll need to set up your machine to run .NET Core, including the C# 8.0 preview compiler. The C# 8 preview compiler is available with [Visual Studio 2019 preview 4](https://visualstudio.microsoft.com/vs/preview/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link&utm_content=download+vs2019+preview), or [.NET Core 3.0 preview 3](https://dotnet.microsoft.com/download/dotnet-core/3.0).
+
+This tutorial assumes you're familiar with C# and .NET, including either Visual Studio or the .NET Core CLI.
+
+## Scenarios for pattern matching
+
+Modern development often includes integrating data from multiple sources and presenting information and insights from that data in a single cohesive application. You and your team won't have control or access for all the types that represent the incoming data.
+
+The classic object-oriented design would call for creating types in your application that represent each data type from those multiple data sources. Then, your application would work with those new types, build inheritance hierarchies, create virtual methods, and implement abstractions. Those techniques work, and sometimes they are the best tools. Other times you can write less code. You can write more clear code using techniques that separate the data from the operations that manipulate that data.
+
+In this tutorial, you'll create and explore an application that takes incoming data from several external sources for a single scenario. You'll see how **pattern matching** provides an efficient way to consume and process that data in ways that weren't part of the original system.
+
+Consider a major metro area that is using tolls and peak time pricing to manage traffic. You write an application that calculates tolls for a vehicle based on its type. Later enhancements incorporate pricing based on the number of occupants in the vehicle. Further enhancements add pricing based on the time and the day of the week.
+
+From that brief description, you may have quickly sketched out an object hierarchy to model this system. If you're building the entire system, and these were the only requirements, that would be a good solution. That's not the situation. This metro area had already been using software to manage vehicle registrations for years. They bought different systems from different suppliers for personal cars, taxis, buses, and delivery trucks. Different systems all provide classes to represent the data they export. But these different systems don't provide a single object model for you to use. In this tutorial, you'll use these simplified classes for the vehicle classes from these external systems, as shown in the following code:
```

Comment:
I had to read this a few times just to make sure I understood what you were trying to say. I think this paragraph could be simplified. there is a lot of text to really just state what is stated a few paragraphs below: 

>You can see that the vehicle classes are from different systems, and are in different namespaces. No common base class, other than `System.Object` can be leveraged.

Also, it is said above this paragraph:

>In this tutorial, you'll create and explore an application that takes incoming data from several external sources for a single scenario. 

So I think that this paragraph is just explaining too much. How about the following or something like it?

```suggestion
From that brief description, you may have quickly sketched out an object hierarchy to model this system. However, your data is coming from multiple sources like other vehicle registration management systems. These systems provide different classes to model that data and you don't have a single object model you can use. In this tutorial, you'll use these simplified classes to model for the vehicle data from these external systems, as shown in the following code:
```